### PR TITLE
Remove SymbolId - Option3

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide Review please.  This is an alternate approach to https://github.com/dotnet/roslyn/pull/11571

Like with that PR, we remove SymbolId and use SymbolKey uniformly across our code.  However, in this impl SymbolKey is represented internally with nothing more than a string (instead of a rich object model in memory).  Operations on it (like Resolving) operate by streaming over that string.

**Comment by CyrusNajmabadi** [comment]
retest vsi please

File: src/Workspaces/Core/Portable/SymbolId/SymbolKey.Visitor.cs [hunk 1]
```diff
@@ -1,140 +1,407 @@
...
 
-            public Visitor(Compilation compilation, CancellationToken cancellationToken)
+            // Not to be confused with ArrayType.  This indicates an array of elements in the stream.
+            Array = '%',
+            Reference = '#',
+            Null = '!',
+            TypeParameterOrdinal = '@',
+        }
+
+        private class Visitor : SymbolVisitor<object>, IDisposable
```
Comment (mattwar) [inline src/Workspaces/Core/Portable/SymbolId/SymbolKey.Visitor.cs]: Its odd now that this is named Visitor, when it is really doing the job of a Serializer.  Visiting is just how it works.

**Pull request closed without merging**